apps/dot_ext/scopes.py

```diff
@@ -21,14 +20,48 @@ def get_available_scopes(self, application=None, request=None, *args, **kwargs):
         Returns a list that contains all the capabilities related
         to the current application.
         """
-        app_scopes = list(
-            ProtectedCapability.objects.filter(Q(default=True) | Q(application=application)).values_list('slug', flat=True))
-        return app_scopes
+        # Return empty scopes list, if application is none
+        if application is None:
+            return []
+
+        # Get list of all available scopes
+        app_scopes_avail = list(
+            ProtectedCapability.objects.filter(Q(default=True) | Q(application=application))
+                                       .values_list('slug', flat=True).distinct())
+
+        # Set scopes based on application choice. Default behavior is True, if it hasn't been set yet.
+        if application.require_demographic_scopes in [True, None]:
```

Comment:
Is it actually possible to meet `application.require_demographic_scopes == None`?  If an application is created by a sandbox user the form will require they select an option here.

Reply:
@njdister Yes.  For existing applications, the value will be Null/None until it is edited or for newly created applications.

Here's the field from the models.py for reference: 
```python
    # Does this application need to collect beneficary demographic information? YES = True/Null NO = False
    require_demographic_scopes = models.BooleanField(default=True, null=True,
                                                     verbose_name="Are demographic scopes required?")
```

Reply:
Yeah you're right about existing applications, that makes sense.  As far as I can tell you cannot create a new application where this would be None since the field is required in order to save on the "Register a New Application" page.